### PR TITLE
[NO GBP] simplifying fishing line shitcode.

### LINKED ISSUE #78357: Fishing exploit is **STILL** present
Reporting client version: 514.1589

## Round ID:
[214637](https://scrubby.melonmesa.com/round/214637)

## Testmerges:
- [[Fuck Shotties] Old Yellers Old Yeller (Improvised Shell Nerf)](https://www.github.com/tgstation/tgstation/pull/78235)

## Reproduction:
get a fishing rod, place pka on floor, rod it, make someone take it, rod it out of their hands, woila! still works. you had one job.

https://github.com/tgstation/tgstation/assets/71607168/64159d56-a70f-4b28-8938-0158e9b73226

dissapointment

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
This should also fix the hooked item exploit for real now, but to avoid further goofs, I'm keeping this as a draft until I finish running the game locally.

## Why It's Good For The Game
Closes #78357.

## Changelog

:cl:
fix: Actually fixed the hooked item exploit.
/:cl:

### CODE REVIEW

File: code/modules/fishing/fishing_rod.dm [hunk 1]
```diff
@@ -66,9 +63,10 @@
 	return NONE
 
 /obj/item/fishing_rod/Destroy(force)
-	. = ..()
-	//Remove any leftover fishing lines
-	QDEL_LIST(fishing_lines)
+	bait = null
+	line = null
+	hook = null
```
Comment (ZephyrTFA) [inline code/modules/fishing/fishing_rod.dm:68]: shouldn't you be qdel'ing these

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/modules/fishing/fishing_rod.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/fishing/fishing_rod.dm:68]: Because they're going to be qdeleted anyway, being in its contents.

Comment (ZephyrTFA) [reply code/modules/fishing/fishing_rod.dm:68]: I'd prefer if(!isnull(x))

Comment (ZephyrTFA) [reply code/modules/fishing/fishing_rod.dm:68]: qdel null will end up qdeling null if the var is already null

Comment (Ghommie) [reply code/modules/fishing/fishing_rod.dm:68]: qdeling null isn't an issue last I checked, it just early returns.

**Review by Jacquerel — CHANGES_REQUESTED** [review]

File: code/modules/fishing/fishing_minigame.dm [hunk 2]
```diff
@@ -130,12 +137,9 @@
 	return COMSIG_MOB_CANCEL_CLICKON
 
 /// Challenge interrupted by something external
-/datum/fishing_challenge/proc/interrupt(datum/source, balloon_alert = TRUE)
-	SIGNAL_HANDLER
+/datum/fishing_challenge/proc/interrupt(datum/source)
```
[the remark this answers is not in the document]
Comment (ninjanomnom) [reply code/modules/fishing/fishing_minigame.dm:140]: Signal handlers always receive the source argument

[the remark this answers is not in the document]
Comment (Jacquerel) [reply code/modules/fishing/fishing_minigame.dm:140]: True, however I think this proc is no longer a signal handler.

**Pull request merged**